apps/mobile/src/components/browser/approval-ux-modal.tsx

```diff
@@ -0,0 +1,80 @@
+import { Ref, useEffect, useRef } from 'react';
+import WebView from 'react-native-webview';
+
+import { formatAddressesForGetAddresses } from '@/hooks/get-addresses';
+import { getDummyKeys } from '@/state/dummy';
+import { BottomSheetModal } from '@gorhom/bottom-sheet';
+
+import {
+  getNativeSegWitPaymentFromAddressIndex,
+  getTaprootPaymentFromAddressIndex,
+} from '@leather.io/bitcoin';
+import { Box, Button } from '@leather.io/ui/native';
+
+import { Modal } from '../bottom-sheet-modal';
+import { BrowserMessage } from './browser-in-use';
+
+interface ApprovalUXModalProps {
+  webViewRef: Ref<WebView>;
+  message: BrowserMessage;
+  setMessage(newMessage: BrowserMessage): void;
+}
+
+export function ApprovalUXModal(props: ApprovalUXModalProps) {
```

Comment:
Think we should avoid calling ApproverUX in code in favour of the "Approver" pattern